turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -607,7 +612,9 @@ pub(crate) async fn analyse_ecmascript_module_internal(
                     }
                     ImportedSymbol::Symbol(name) => Some(ModulePart::export((&**name).into())),
                     ImportedSymbol::PartEvaluation(part_id) => {
-                        evaluation_references.push(i);
+                        if !is_side_effect_free {
```

Comment:
Without this change, all next.js tests pass (CI) but the side-effect-optimization unit test fails because of `non-existing.js`.